### ISSUE TITLE
Test Set Used During Training Loop

### ISSUE THREAD

**Issue by AI4Science2025** [junxia97/Mole-BERT#32]
Hi, thanks for sharing this great repo — it’s been very helpful for my research.
I noticed a potential problem in how the train/validation/test sets are used in finetune.py:
`for epoch in range(1, args.epochs+1):
    train(args, epoch, model, device, train_loader, optimizer)

    val_acc = eval(args, model, device, val_loader)
    test_acc = eval(args, model, device, test_loader)   # <-- called every epoch`
**Concern**
	•	In the current implementation, the test set is evaluated at every epoch, just like validation.
	•	This means the test set is being used as part of the training feedback loop.
	•	However, the test set is supposed to be a final hold-out set, evaluated only once after model selection (based on validation).
	•	Repeatedly checking test accuracy during training can lead to test set leakage: the model may indirectly “tune” itself to the test data, and the reported test performance will be optimistically biased.
**Suggested Fix**
	•	Use validation accuracy to monitor training and select the best checkpoint.
	•	After training completes, evaluate the saved best model on the test set only once.
